### DISCUSSION TITLE
Workflow 2x2U mapping choice not enforced

### DISCUSSION THREAD

**Discussion by nwhelan** [manna-harbour/miryoku#155] [Q&A] [answered]
I am using GitHub Actions to build firmware for a [Technikable with a 2x2U layout](https://github.com/manna-harbour/miryoku_zmk#2x2u) which should result in the inner-most thumb keys being extended, while all other thumb keys remain unmodified. However, when flashed to the board, all thumb keys keep the unmodified layout.

I have reviewed the mapping file and [thought to try updating it](https://github.com/nwhelan/miryoku_zmk/blob/master/miryoku/mapping/44/technikable.h), though that has been unsuccessful as well.

[Here is the workflow file I am using to build with.](https://github.com/nwhelan/miryoku_zmk/blob/master/.github/workflows/technikable.yml) Thanks in advance for your help!

**Comment by manna-harbour** [comment]
I'm just stepping out, but the short answer is to see https://github.com/manna-harbour/miryoku_zmk#miryoku-alternative-layout-and-mapping-options and edit https://github.com/nwhelan/miryoku_zmk/blob/07e9c0f9a2a56131c34713b0ec9cfe7df1bd6c9a/.github/workflows/technikable.yml#L12 to remove the prefix.

**Reply by nwhelan** [reply]
Thank you, Manna! Somehow ended up trying to use the config file option within the mapping field, I guess. Appreciate your time and a big fan of your work here.

**Comment by manna-harbour** [comment, marked as answer]
At https://github.com/nwhelan/miryoku_zmk/blob/07e9c0f9a2a56131c34713b0ec9cfe7df1bd6c9a/.github/workflows/technikable.yml#L12 you have `mapping: '["MIRYOKU_MAPPING_2X2U"]'`.

For workflow alternative layout and mapping options, see https://github.com/manna-harbour/miryoku_zmk#miryoku-alternative-layout-and-mapping-options. In particular

> Alternative layout and mapping options are given in the documentation in the form MIRYOKU_OPTION=VALUE, e.g. MIRYOKU_ALPHAS=QWERTY. To use here, use the value with the corresponding option.

So, even though the end result is a macro called `MIRYOKU_MAPPING_2X2U`, you need to use only `2X2U` with the workflow `mapping` option.

That is a bit confusing though, as if you want to set defaults in the config file, you'd use `#define MIRYOKU_MAPPING_2X2U`. See https://github.com/manna-harbour/miryoku_zmk#config-file.

At https://github.com/nwhelan/miryoku_zmk/blob/3c18cd358059e42c6abe9751e7eab19bdc1b7d17/.github/workflows/technikable.yml#L12 you have `custom_config: '["#define MIRYOKU_ALPHAS_QWERTY\n#define MIRYOKU_EXTRA_COLEMAKDH\n#define MIRYOKU_TAP_QWERTY\n#define MIRYOKU_MAPPING_2X2U"]'`. That will work but it might be a bit hard to manage. For `alphas` and `mapping` you can use the corresponding workflow option instead. For Extra and Tap there is no workflow option and you need to use `custom_config` as you've done. Alternatively, set all your defaults in the config file.
